qemu/tests/edk2_basic.py

```diff
@@ -0,0 +1,39 @@
+import time
+
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu edk2 basic test:
+    1) Try to log into a guest
+    2) Check serial log information
+    3) Check edk2 output information
+    4) Add iommu test scenario
+    5) Cycle the above process
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    timeout = params.get_numeric("login_timeout", 240)
+    check_messgae = params.get("check_messgae")
+    vm = env.get_vm(params["main_vm"])
+
+    for i in range(params.get_numeric("reboot_count", 1)):
+        vm.create()
+        error_context.context("Check serial log result", test.log.info)
+        try:
+            vm.serial_console.read_until_output_matches([check_messgae],
+                                                        timeout=timeout)
+            error_context.context("Check edk2 output information", test.log.info)
+            output = vm.serial_console.get_output()
```

Comment:
`read_until_output_matches` will return output, and the final line matches the `[check_message]`, better than getting new data from get_output.

Reply:
The output here is to capture all logs to match the number of lines. The keyword matching has been used above with `read_until_output_matches`

Reply:
I don't understand re-getting the output here. the output of `read_until_output_matches` returns lines before `check_message`, serial log prints very quickly,  `vm.serial_console.get_output()` may get more lines after `check_message`, which is not accurate to use to judge.

Reply:
Corrected